extensions/widgets/treeview/treeview.lcb

```diff
@@ -1511,7 +1511,7 @@ private handler addKey(in pListElt as Integer, in pPath as List, in pLevel as In
 	else
 		variable tArray as Array
 		put the empty array into tArray	
-		put "" into tArray[1 formatted as string]
+		put tElement into tArray[1 formatted as string]
```

Comment:
@bwmilby I wonder if rather than move the element which might be hard to recover from it should check if `tElement` is empty or not and post an error message if it's not. That way script can handle the message and ask the user if they want to move the current value to an array element. It could be they just clicked `+` on the wrong line and would like to back out.

Reply:
We could I guess add the reverse of this behavior so it's easy to back out but it's not all that intuitive to click the bin on an array element in an array with one key and have the value clobber the array.

Reply:
I considered the error route (similar to the delete message) but thought this would be less intrusive for applications.  The problem about the reverse is you have a key and a value to deal with so you will lose one anyway.

Reply:
Hmm... yeah the way the delete message is done would work rather than a callback. I imagine the main choice people would want to make is between canceling and replacing the element with an array. Using the current element value as array key 1 is possibly more of an edge case... not sure. I guess it could be a third button on the dialog...

```
answer warning "The array element at path [" & <path> & "] is not empty. Would you like to replace it with an empty array or move it to the first element?" with "Cancel" or "Replace" or "Move"
```

Reply:
Great thanks @bwmilby